src/utils/index.ts

```diff
@@ -617,6 +617,32 @@ export async function shellWithDocker(): Promise<vscode.Terminal> {
   return terminal;
 }
 
+interface WSServerRootFolderData {
+  redirectDotvscode: boolean;
+}
+
+const wsServerRootFolders = new Map<string, WSServerRootFolderData>();
+
+/**
+ * Add uri to the wsServerRootFolders map if eligible
+ */
+export async function addWsServerRootFolderData(uri: vscode.Uri): Promise<void> {
+  if (!schemas.includes(uri.scheme)) {
+    return;
+  }
+  const value: WSServerRootFolderData = {
+    redirectDotvscode: true,
+  };
+  if (isCSPFile(uri)) {
+    // A CSP-type root folder that already has a .vscode/settings.json file must not redirect .vscode/* references
+    const api = new AtelierAPI(uri);
+    api.getDoc(`${uri.path}/.vscode/settings.json`).then(() => {
```

Comment:
This should use `headDoc` instead, and needs a `catch`.

Reply:
Done